### PR TITLE
Allow attaching service pods to additional networks

### PR DESCRIPTION
**Pull request by stuggi**
For network isolation outgoing traffic, service pods need to be attached additional networks. To achive this, the the `k8s.v1.cni.cncf.io/networks` annotation needs to be set on the pods. This adds a networkAttachmentDefinitions parameter to the CRD where it can be specified to which additional networks the service pods should be attached to.

Depends-On: https://github.com/openstack-k8s-operators/lib-common/pull/133

### CODE REVIEW

**Review by abays — COMMENTED** [review]
+1 from me.  Are we still holding this?

File: go.mod [hunk 1]
```diff
@@ -92,3 +92,5 @@ replace github.com/openstack-k8s-operators/keystone-operator/api => ./api
 // Without this, the following error occurs:
 // ../go/pkg/mod/k8s.io/apimachinery@v0.24.3/pkg/util/managedfields/gvkparser.go:62:39: cannot use smdschema.Schema{…} (value of type "sigs.k8s.io/structured-merge-diff/v4/schema".Schema) as type *"sigs.k8s.io/structured-merge-diff/v4/schema".Schema in struct literal
 replace sigs.k8s.io/structured-merge-diff/v4 v4.2.2 => sigs.k8s.io/structured-merge-diff/v4 v4.2.1
+
+replace github.com/openstack-k8s-operators/lib-common/modules/common => github.com/stuggi/lib-common/modules/common v0.0.0-20230113114514-f193a36429c0
```
Comment (olliewalsh) [inline go.mod:96]: Was onhold waiting for this to merge?

Comment (stuggi) [reply go.mod:96]: yes, I'll update it today

**Comment by stuggi** [comment]
`go.mod` updated. Also added the network list to the status and to have it printed on the cli summary, like:

```
$ oc get keystoneapi
NAME       NETWORKS                    STATUS   MESSAGE
keystone   ["internalapi","storage"]   True     Setup complete
```

**Review by olliewalsh — APPROVED** [review]
/lgtm

**Review by abays — APPROVED** [review]

**Pull request merged**